lib/thrift/binary/framed/server.ex

```diff
@@ -2,28 +2,31 @@ defmodule Thrift.Binary.Framed.Server
   do
   @moduledoc false
 
-  @type server_opts :: [
-    worker_count: pos_integer,
-    name: atom,
-    max_restarts: non_neg_integer,
-    max_seconds: non_neg_integer,
-    tcp_opts: :ranch_tcp.opts
-  ]
+  @type server_option ::
+    {:worker_count, pos_integer} |
+    {:name, atom} |
+    {:max_restarts, non_neg_integer} |
+    {:max_seconds, non_neg_integer} |
+    {:tcp_opts, :ranch_tcp.opts} |
+    {:ssl_opts, [Thrift.Transport.SSL.option]}
 
-  @spec start_link(module, (1..65_535), module, server_opts) :: GenServer.on_start
+  @type server_opts :: [server_option]
+
+  @spec start_link(module, (1..65_535), module, [server_option]) :: GenServer.on_start
```

Comment:
Typespec should use `server_opts` instead of `[server_option]`

Reply:
Elements of a list should be the types to allow composing. Have a list type for options is an anti pattern because it does not allow combining with other types. I changed this type issue in all the nearby code affect by this PR to use [option] style.

Reply:
I only left the `_opts` type for backwards compatibility.